index.js

```diff
@@ -22,83 +22,407 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-import sh from 'execSync';
+//import sh from 'execSync';
+import i2c from 'i2c-bus';
+import { execSync } from 'child_process';
 import { Peripheral } from 'raspi-peripheral';
+import { VERSION_1_MODEL_B_REV_1, getBoardRevision } from 'raspi-board';
+
+if (typeof execSync !== 'function') {
+  execSync = require('execsync');
+}
+
+function checkAlive(alive) {
+  if (!alive) {
+    throw new Error('Attempted to access a destroyed I2C peripheral');
+  }
+}
+
+function checkAddress(address) {
+  if (typeof address !== 'number' || address < 0 || address > 0x7f) {
+    throw new Error('Invalid I2C address ' + address
+      + '. Valid addresses are 0 through 0x7f.'
+    );
+  }
+}
+
+function checkRegister(register) {
+  if (register !== undefined &&
+      (typeof register !== 'number' || register < 0 || register > 0xff)) {
+    throw new Error('Invalid I2C register ' + register
+      + '. Valid registers are 0 through 0xff.'
+    );
+  }
+}
+
+function checkLength(length) {
+  if (typeof length !== 'number' || length < 0 || length > 32) {
+    throw new Error('Invalid I2C length ' + length
+      + '. Valid lengths are 0 through 32.'
+    );
+  }
+}
+
+function checkCallback(cb) {
+  if (typeof cb !== 'function') {
+    throw new Error('Invalid I2C callback ' + cb);
+  }
+}
+
+function checkBuffer(buffer) {
+  if (!Buffer.isBuffer(buffer) || buffer.length <= 0 || buffer.length > 32) {
+    throw new Error('Invalid I2C buffer ' + buffer
+      + '. Valid lengths are 0 through 32.'
+    );
+  }
+}
+
+function checkByte(byte) {
+  if (typeof byte !== 'number' || byte < 0 || byte > 0xff) {
+    throw new Error('Invalid I2C byte ' + byte
+      + '. Valid values are 0 through 0xff.'
+    );
+  }
+}
+
+function checkWord(word) {
+  if (typeof word !== 'number' || word < 0 || word > 0xffff) {
+    throw new Error('Invalid I2C word ' + word
+      + '. Valid values are 0 through 0xffff.'
+    );
+  }
+}
+
+var devices = '__r$396836_0$__';
+var getDevice = '__r$396836_1$__';
 
 export class I2C extends Peripheral {
   constructor(baudRate, pins) {
```

Comment:
Need to change the signature to match what we talked about in https://github.com/bryan-m-hughes/raspi-io/issues/16